consensus/state.go

```diff
@@ -720,11 +720,7 @@ func (cs *ConsensusState) needProofBlock(height int64) bool {
 func (cs *ConsensusState) proposalHeartbeat(height int64, round int) {
 	counter := 0
 	addr := cs.privValidator.GetAddress()
-	valIndex, v := cs.Validators.GetByAddress(addr)
-	if v == nil {
-		// not a validator
-		valIndex = -1
-	}
+	valIndex, _ := cs.Validators.GetByAddress(addr)
```

Comment:
what happened here ? Looks like GetByAddress returns 0 if its not found, not -1 ? 

Also noticing it only checks `idx != len(valSet.Validators)` but should probably use `<` ?

Reply:
Oh awesome I see you fixed this. Thanks!